coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/functions/HiveReturnTypes.java

```diff
@@ -61,6 +67,40 @@ private HiveReturnTypes() {
     }
   };
 
+  /**
+   * Represents the return type for the coalesce_struct UDF that is built for bridging the schema difference
+   * between previous Coral's interpretation of union field in Coral IR (i.e. consistent to the output schema of
+   * extract_union Hive UDF, let's call it struct_old) and Trino's schema when deserializing union field from its reader.
+   * (Let's call it struct_new, See https://github.com/trinodb/trino/pull/3483 for details).
+   *
+   * struct_new looks like:
+   * struct&lt;tag:int, field0:type0, field1:type1, ... fieldN:typeN&gt;
+   *
+   * struct_old looks like:
+   * struct&lt;tag_0:type0, tag_1:type1, ... tag_N:typeN&gt;
+   *
+   * This new UDF could be stated as the following signature:
```

Comment:
`The `coalesce_struct` function signature can be expressed as follows:`
Also I would separate to two signatures.